apps/nvml.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apps
+
+import (
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+	"github.com/GoogleCloudPlatform/ops-agent/internal/platform"
+)
+
+type MetricsReceiverNvml struct {
+	confgenerator.ConfigComponent       `yaml:",inline"`
+	confgenerator.MetricsReceiverShared `yaml:",inline"`
+}
+
+func (r MetricsReceiverNvml) Type() string {
+	return "nvml"
```

Comment:
I thought we were going to use a more meaningful name for the receiver? (Like `nvidia_gpu` or `nvidia_nvml` or something)

Reply:
N/A anymore - merging the nvml receiver into hostmetrics.